x-pack/plugins/enterprise_search/public/applications/shared/flash_messages/flash_messages_logic.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea } from 'kea';
+import { ReactNode } from 'react';
+import { History } from 'history';
+
+import { IKeaLogic, TKeaReducers, IKeaParams } from '../types';
+
+export interface IFlashMessage {
+  type: 'success' | 'info' | 'warning' | 'error';
```

Comment:
consider using an [enum](https://www.typescriptlang.org/docs/handbook/enums.html)!

Reply:
I don't super mind if we're all on board as a team, since it'll change our current flash messages usage. We'll need to always import the enum and do something like

`setFlashMessages({ type: TFlashMessageType.SUCCESS, message: 'hello world' });`
on every usage to pass Typescript's checks, vs.
vs. just
`setFlashMessages({ type: 'success', message: 'hello world' });` (current usage)

Which, I mean, I know it's "technically" better(?), but is it really all that worth it? 🤷‍♀️ Also worth considering that [EUI's typing itself](https://github.com/elastic/eui/blob/master/src/components/call_out/call_out.tsx#L29) is a string literal and not an enum.

CC @scottybollinger @JasonStoltz @yakhinvadim - feel free to chime in if I'm alone in my indifference to string enums (I think they make more sense/have more value for number enums and other types though)

Reply:
Ah I see that this is modeling EUI (I wish they exported that type!!!!).  In that case this is fine by me